Averaging to adding

JennaGustafson opened the pull request.

Commit:
prints mof array with lowest kld value as a result

Commit:
plots pdf values insead of discrete probabilities

Commit:
plots smooth line using spline function

Commit:
saves results to csv, ordered list of arrays/gas/kld

Commit:
plot only scatter pdf, set axis number labels to font size 20

Commit:
truncated normal prob dist, lower limit of 0 mass

Commit:
added function to save probablity data for each array/gas combo

Commit:
takes product of KLD values for an array over 3 gases,then ranks performance

Commit:
saves pmf data to csv for individual mofs

Commit:
conditional, only takes kld products for ternary gas mixtures

Commit:
add bins instead of average

Commit:
calculate joint pmf before binning points

Commit:
squash kld stuff

Commit:
calculates kld product for 3+ component mixtures, chooses top/bottom two

Commit:
ranks top/bottom two for mixts with 2 gases

Commit:
extract top and bottom arrays overall and by gas

Commit:
updated comments

Commit:
removed unused variable

Commit:
returns complete list of arrays, ranked

Commit:
separated lines defining variables and those saving/plotting data

Commit:
updated function argument definitions

Commit:
saves array pdfs for all arrays, fixed indenting

Commit:
binary gas mixtures, joint KLD is just kld of one gas, fixed

paulboone commented on execute_process_mass_data.py:
```diff
@@ -49,7 +49,7 @@
 # save_array_pmf_data(gases, list_of_arrays, create_bins_results, bin_compositions_results)
 kl_divergence = information_gain(gases, list_of_arrays, bin_compositions_results, create_bins_results)
-combined_kld, ordered_by_gas, best_arrays, ordered_kld_w_array, average_kld = choose_best_arrays(gases, kl_divergence)
+# combined_kld, ordered_by_gas, best_arrays, ordered_kld_w_array, average_kld = choose_best_arrays(gases, kl_divergence)
```
commented out code

paulboone commented on execute_process_mass_data.py:
```diff
@@ -49,7 +49,7 @@
 # plot_binned_pmf_array(gases, list_of_arrays, create_bins_results, bin_compositions_results)
```
generally the word "squash" in a commit message indicates that you forgot to actually squash

paulboone commented on execute_process_mass_data.py:
```diff
@@ -41,28 +41,28 @@
-for num_mofs in range(0,len(average_kld)):
-    print('\t%s MOF array is %s' %(num_mofs + 1, average_kld[num_mofs]))
+# write_output_data('saved_results/ordered_by_kld_product_%s.csv' % (datetime.now().strftime("%Y_%m_%d__%H_%M_%S")), combined_kld)
+# write_output_data('saved_results/ordered_by_gas_kld_%s.csv' % (datetime.now().strftime("%Y_%m_%d__%H_%M_%S")), ordered_by_gas)
```
commented out code

paulboone reviewed (changes requested):
Looks good. You had some commenting in / out of code which should never really be committed. Can you check those and make sure they were actually removed?

JennaGustafson commented:
There were two lines of code commented out in the final version of process_mass_data.py, so I deleted them (they are not used) and squashed that commit further back where I initially committed the commented out code. The final version looks better.
